lib/ReverseGeocode.php

```diff
@@ -107,12 +109,53 @@ function lookup()
 				}
 				$iPlaceID = $aPlace['place_id'];
 				$iParentPlaceID = $aPlace['parent_place_id'];
+				$bIsInUnitedStates = ($aPlace['calculated_country_code'] == 'us');
+			}
+
+
+			// Only street found? If it's in the US we can check TIGER data for nearest housenumber
+			if ($bIsInUnitedStates && $iPlaceID && $aPlace['rank_search'] == 26) 
```

Comment:
rank_search == 27 might be relevant, too (named service roads, tracks and footways)